app/routers/quizzes.py

```diff
@@ -39,19 +44,42 @@ def update_quiz_for_backwards_compatibility(quiz_collection, quiz_id, quiz):
                     "skipped": 0,
                 }  # default
 
-    quiz_collection.update_one({"_id": quiz_id}, {"$set": quiz})
+    if is_backwards_compatibile:
+        logger.info("Quiz is already backwards compatible")
+        return
+
+    logger.info("Starting update for backwards compatibility")
+    update_result = quiz_collection.update_one({"_id": quiz_id}, {"$set": quiz})
+
+    if not update_result.acknowledged:
+        logger.error("Failed to update quiz for backwards compatibility")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Failed to update quiz for backwards compatibility",
+        )
+
+    logger.info("Quiz updated for backwards compatibility")
 
 
 @router.post("/", response_model=CreateQuizResponse)
 async def create_quiz(quiz: Quiz):
+    logger.info("Starting quiz creation")
```

Comment:
can we have some description for quiz? 

we could use quiz source id for now. its in metadata
we should do this after `jsonable_encoder(quiz)` step

if "metadata" in quiz and "source-id" in quiz["metadata"]:
    logger.info("starting quiz creation with source id quiz["metadata"]["source_id"]